### DISCUSSION TITLE
Page Access depending on User Role

### DISCUSSION THREAD

**Discussion by sluxzinc92** [filamentphp/filament#9472] [Help] [answered]
Please help me, how to control page access depending on role. For example for users page resources, i wan't it to be accessed only by users with superadmin role. I tried laravel middleware but it's not working. Filament uses routes like magic(just kidding).  Anyway, please somebody help me.

**Comment by VincvanAs** [comment]
For resources you can create a policy for managing access and for custom pages  you can use:
```
public function mount(): void
{   
 abort_unless(auth()->user()->canManageSettings(), 403);
}
```
https://laravel.com/docs/10.x/authorization#generating-policies

https://filamentphp.com/docs/3.x/panels/pages#conditionally-hiding-pages-in-navigation

**Comment by sluxzinc92** [comment]
Sorry but resources are handled in filament differently. I'm hoping there could be a code in userresource.php where it will limit the access of the user depending on its role.

As for hiding custom pages, it's not a custom page. And for hiding navigation, the page is still available when directly opening the link.

**Reply by VincvanAs** [reply, marked as answer]
You can handle resources by policies. If you set the viewany method to false it wont be shown in the navigation and it cannot be accessed by the user

**Reply by chris-john19** [reply]
how do you do that for multiple user roles? like this one can only view and that one can edit. I understand we can use spatie...but can you give examples how to use spatie in filament? dont have any experience also with spatie but I'm currently reading the documentation. I wonder how that can be done with filament

**Reply by VincvanAs** [reply]
In your policy you define multiple functions, like create, delete, view, update etc.

Then you can use the spatiepermissions e.g.

```
$user->can('Permission name');
```

Or role e.g.

```
$user->hasRole('Role name');
```



```
<?php

namespace App\Policies;

use App\Models\User;

class UserPolicy
{
    /**
     * Determine whether the user can view any models.
     */
    public function viewAny(User $user): bool
    {
        return $user->can('View Users');
    }

    /**
     * Determine whether the user can view the model.
     */
    public function view(User $user, User $model): bool
    {
        return $user->can('View User');
    }

    /**
     * Determine whether the user can create models.
     */
    public function create(User $user): bool
    {
        return $user->can('Add Users');
    }

    /**
     * Determine whether the user can update the model.
     */
    public function update(User $user, User $model): bool
    {
        return $user->can('Edit Users');
    }

    /**
     * Determine whether the user can delete the model.
     */
    public function delete(User $user, User $model): bool
    {
        return $user->can('Delete Users');
    }

    /**
     * Determine whether the user can restore the model.
     */
    public function restore(User $user, User $model): bool
    {
      //
    }

    /**
     * Determine whether the user can permanently delete the model.
     */
    public function forceDelete(User $user, User $model): bool
    {
      //
    }

}
```
